chef/cookbooks/provisioner/recipes/base.rb

```diff
@@ -70,7 +70,8 @@
 
 # Find provisioner servers and include them.
 provisioner_server_node = nil
-search(:node, "roles:provisioner-server AND provisioner_config_environment:#{node[:provisioner][:config][:environment]}") do |n|
+provisioners = node_search_with_cache("roles:provisioner-server")
```

Comment:
is this search equivalent with the one that it replaces? we had more conditions there. Again, I might lack some very basic chef knowledge here.